### ISSUE TITLE
Refine lattice data encapsulation based on review

### ISSUE THREAD

**Issue by schmie** [schmie/hemeMove#197]
**Reported by jamespjh on 7 Mar 2012 20:27 UTC**
See the review in wiki:GeometryReview170112

**Comment by schmie** [comment]
**Modified by hywel on 8 Mar 2012 14:08 UTC**

**Comment by schmie** [comment]
**Modified by hywel on 26 Apr 2012 12:36 UTC**

**Comment by schmie** [comment]
**Modified by hywel on 26 Apr 2012 12:36 UTC**

**Comment by schmie** [comment]
**Comment by hywel on 26 Apr 2012 12:57 UTC**
Small classes, testing with expected communications paradigm like the !IncrompressibilityTester tests.

**Comment by schmie** [comment]
**Comment by hywel on 11 May 2012 14:35 UTC**
af6081e4e2fd
864e73c96140
2cc9779929f2
65dd57f309da
88b296225ad4
24479faac89c
b73450cb722b
c263ed287661

These are all in response to things you've said in your big review, so feel free to not do this properly.

Anything I haven't fixed is either out of date, or has had a new ticket raised for it. This includes the unit tests mentioned in comment:4

**Comment by schmie** [comment]
**Modified by jamespjh on 14 May 2012 12:46 UTC**

**Comment by schmie** [comment]
**Modified by jamespjh on 28 May 2012 14:27 UTC**

**Comment by schmie** [comment]
**Modified by jamespjh on 28 May 2012 14:55 UTC**

**Comment by schmie** [comment]
**Comment by jamespjh on 28 May 2012 15:37 UTC**
Only two comments:

* Make !ShouldValidate be a separate CMAKE parameter, rather than going off debug
* Near !SendAndReceive and !CopyReceived, add commenting to explain the comms process - this is a little obscure

Really nice cleanup, I appreciate it! Thanks Hywel.

**Comment by schmie** [comment]
**Modified by jamespjh on 28 May 2012 15:37 UTC**

**Comment by schmie** [comment]
**Comment by jamespjh on 28 May 2012 15:38 UTC**
Removing my estimate for doing the review.

**Comment by schmie** [comment]
**Comment by hywel on 29 May 2012 11:16 UTC**
Took me a while but done in e7df86b2dfc4. I think it's big enough to need reviewing (and I'm no CMake expert so would prefer someone looked at the stuff around that).

**Comment by schmie** [comment]
**Comment by jamespjh on 30 May 2012 09:33 UTC**
The option from the entry CMake file needs to be added to the list of options passed to the subproject -- the long list of repetitve -DFOO=${FOO}s

Since the option in CMake is ON/OFF, the add definition line should be 
```
if(HEMELB_VALIDATE_GEOMETRY)
add_definition(HEMELB_VALIDATE_GEOMETRY)
endif()
```
-- otherwise I think it will always take the defined route (since defined as OFF is still defined -- though there could be a subtlety in cmake I don't understand.)

Rest fine.

**Comment by schmie** [comment]
**Comment by hywel on 30 May 2012 11:20 UTC**
Good points. Fixed in 3e6ac4a6039c.

Probably don't need to review it
